### ISSUE TITLE
NullpointerException after Docker upgrade

### ISSUE THREAD

**Issue by tstroband** [spotify/dockerfile-maven#33]
Hi,

Just after upgrading Docker Community Edition on OSX (Sierra 10.12.5) to Version 17.06.0-ce-mac18 (18433) Channel: stable d9b66511e0 I got this NullPointerException when building my project.

Any ideas how I can fix this?

```
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T17:41:47+01:00)
Maven home: /Applications/IntelliJ IDEA.app/Contents/plugins/maven/lib/maven3
Java version: 1.8.0_60, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.8.0_60.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.12.5", arch: "x86_64", family: "mac"
...
[INFO] --- dockerfile-maven-plugin:1.3.0:build (default) @ myproject ---
[DEBUG] Using mirror dvp-nexus (http://dvprepo.dvg.local/dvpnexus/content/groups/public) for jcenter (http://jcenter.bintray.com).
[DEBUG] Using mirror dvp-nexus (http://dvprepo.dvg.local/dvpnexus/content/groups/public) for jvnet-nexus-releases (https://maven.java.net/content/repositories/releases/).
[DEBUG] Using mirror dvp-nexus (http://dvprepo.dvg.local/dvpnexus/content/groups/public) for jvnet-nexus-snapshots (https://maven.java.net/content/repositories/snapshots).
[DEBUG] Using mirror dvp-nexus (http://dvprepo.dvg.local/dvpnexus/content/groups/public) for spring-external (http://repository.springsource.com/maven/bundles/external).
[DEBUG] Using mirror dvp-nexus (http://dvprepo.dvg.local/dvpnexus/content/groups/public) for glassfish-repo-archive (http://maven.glassfish.org/content/groups/glassfish).
[DEBUG] Dependency collection stats: {ConflictMarker.analyzeTime=0, ConflictMarker.markTime=1, ConflictMarker.nodeCount=256, ConflictIdSorter.graphTime=0, ConflictIdSorter.topsortTime=0, ConflictIdSorter.conflictIdCount=95, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=5, ConflictResolver.conflictItemCount=187, DefaultDependencyCollector.collectTime=451, DefaultDependencyCollector.transformTime=6}
[DEBUG] com.spotify:dockerfile-maven-plugin:jar:1.3.0:
[DEBUG]    com.spotify:docker-client:jar:shaded:8.7.1:compile
[DEBUG]       org.slf4j:slf4j-api:jar:1.7.22:compile
[DEBUG]       com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:jar:2.8.8:compile
[DEBUG]          com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:jar:2.8.8:compile
[DEBUG]          com.fasterxml.jackson.core:jackson-core:jar:2.8.8:compile
[DEBUG]          com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.8.8:compile
[DEBUG]       com.fasterxml.jackson.datatype:jackson-datatype-guava:jar:2.8.8:compile
[DEBUG]       com.fasterxml.jackson.core:jackson-databind:jar:2.8.8:compile
[DEBUG]          com.fasterxml.jackson.core:jackson-annotations:jar:2.8.0:compile
[DEBUG]       org.glassfish.jersey.core:jersey-client:jar:2.22.2:compile
[DEBUG]          javax.ws.rs:javax.ws.rs-api:jar:2.0.1:compile
[DEBUG]          org.glassfish.jersey.core:jersey-common:jar:2.22.2:compile
[DEBUG]             javax.annotation:javax.annotation-api:jar:1.2:compile
[DEBUG]             org.glassfish.jersey.bundles.repackaged:jersey-guava:jar:2.22.2:compile
[DEBUG]             org.glassfish.hk2:osgi-resource-locator:jar:1.0.1:compile
[DEBUG]          org.glassfish.hk2:hk2-api:jar:2.4.0-b34:compile
[DEBUG]             org.glassfish.hk2:hk2-utils:jar:2.4.0-b34:compile
[DEBUG]             org.glassfish.hk2.external:aopalliance-repackaged:jar:2.4.0-b34:compile
[DEBUG]          org.glassfish.hk2.external:javax.inject:jar:2.4.0-b34:compile
[DEBUG]          org.glassfish.hk2:hk2-locator:jar:2.4.0-b34:compile
[DEBUG]             org.javassist:javassist:jar:3.18.1-GA:compile
[DEBUG]       org.glassfish.jersey.connectors:jersey-apache-connector:jar:2.22.2:compile
[DEBUG]       org.glassfish.jersey.media:jersey-media-json-jackson:jar:2.22.2:compile
[DEBUG]          org.glassfish.jersey.ext:jersey-entity-filtering:jar:2.22.2:compile
[DEBUG]       org.apache.commons:commons-compress:jar:1.9:compile
[DEBUG]       commons-io:commons-io:jar:2.5:compile
[DEBUG]       org.apache.httpcomponents:httpclient:jar:4.5:compile
[DEBUG]          commons-logging:commons-logging:jar:1.2:compile
[DEBUG]          commons-codec:commons-codec:jar:1.9:compile
[DEBUG]       org.apache.httpcomponents:httpcore:jar:4.4.5:compile
[DEBUG]       com.github.jnr:jnr-unixsocket:jar:0.8:compile
[DEBUG]          com.github.jnr:jnr-ffi:jar:2.0.3:compile
[DEBUG]             com.github.jnr:jffi:jar:1.2.9:compile
[DEBUG]             com.github.jnr:jffi:jar:native:1.2.9:runtime
[DEBUG]             org.ow2.asm:asm:jar:5.0.3:compile
[DEBUG]             org.ow2.asm:asm-commons:jar:5.0.3:compile
[DEBUG]             org.ow2.asm:asm-analysis:jar:5.0.3:compile
[DEBUG]             org.ow2.asm:asm-tree:jar:5.0.3:compile
[DEBUG]             org.ow2.asm:asm-util:jar:5.0.3:compile
[DEBUG]             com.github.jnr:jnr-x86asm:jar:1.0.2:compile
[DEBUG]          com.github.jnr:jnr-constants:jar:0.8.7:compile
[DEBUG]          com.github.jnr:jnr-enxio:jar:0.9:compile
[DEBUG]          com.github.jnr:jnr-posix:jar:3.0.12:compile
[DEBUG]       commons-lang:commons-lang:jar:2.6:compile
[DEBUG]       org.bouncycastle:bcpkix-jdk15on:jar:1.52:compile
[DEBUG]          org.bouncycastle:bcprov-jdk15on:jar:1.52:compile
[DEBUG]       ch.qos.logback:logback-classic:jar:1.2.1:compile
[DEBUG]          ch.qos.logback:logback-core:jar:1.2.1:compile
[DEBUG]    com.google.auth:google-auth-library-oauth2-http:jar:0.6.0:compile
[DEBUG]       com.google.auth:google-auth-library-credentials:jar:0.6.0:compile
[DEBUG]       com.google.http-client:google-http-client:jar:1.19.0:compile
[DEBUG]          com.google.code.findbugs:jsr305:jar:2.0.1:compile
[DEBUG]       com.google.http-client:google-http-client-jackson2:jar:1.19.0:compile
[DEBUG]    com.google.guava:guava:jar:19.0:compile
[DEBUG]    com.spotify:dockerfile-maven-extension:jar:1.3.0:compile
[DEBUG]    org.apache.maven:maven-plugin-api:jar:3.3.9:compile
[DEBUG]       org.apache.maven:maven-model:jar:3.3.9:compile
[DEBUG]       org.apache.maven:maven-artifact:jar:3.3.9:compile
[DEBUG]       org.eclipse.sisu:org.eclipse.sisu.plexus:jar:0.3.2:compile
[DEBUG]          javax.enterprise:cdi-api:jar:1.0:compile
[DEBUG]             javax.annotation:jsr250-api:jar:1.0:compile
[DEBUG]          org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.2:compile
[DEBUG]    org.apache.maven:maven-core:jar:3.3.9:compile
[DEBUG]       org.apache.maven:maven-settings:jar:3.3.9:compile
[DEBUG]       org.apache.maven:maven-settings-builder:jar:3.3.9:compile
[DEBUG]          org.apache.maven:maven-builder-support:jar:3.3.9:compile
[DEBUG]       org.apache.maven:maven-repository-metadata:jar:3.3.9:compile
[DEBUG]       org.apache.maven:maven-model-builder:jar:3.3.9:compile
[DEBUG]       org.apache.maven:maven-aether-provider:jar:3.3.9:compile
[DEBUG]          org.eclipse.aether:aether-spi:jar:1.0.2.v20150114:compile
[DEBUG]       org.eclipse.aether:aether-impl:jar:1.0.2.v20150114:compile
[DEBUG]       org.eclipse.aether:aether-api:jar:1.0.2.v20150114:compile
[DEBUG]       org.eclipse.aether:aether-util:jar:1.0.2.v20150114:compile
[DEBUG]       com.google.inject:guice:jar:no_aop:4.0:compile
[DEBUG]          javax.inject:javax.inject:jar:1:compile
[DEBUG]          aopalliance:aopalliance:jar:1.0:compile
[DEBUG]       org.codehaus.plexus:plexus-interpolation:jar:1.21:compile
[DEBUG]       org.codehaus.plexus:plexus-utils:jar:3.0.22:compile
[DEBUG]       org.codehaus.plexus:plexus-classworlds:jar:2.5.2:compile
[DEBUG]       org.codehaus.plexus:plexus-component-annotations:jar:1.6:compile
[DEBUG]       org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3:compile
[DEBUG]          org.sonatype.plexus:plexus-cipher:jar:1.4:compile
[DEBUG]       org.apache.commons:commons-lang3:jar:3.4:compile
[DEBUG]    org.apache.maven:maven-archiver:jar:3.0.0:compile
[DEBUG]       org.apache.maven.shared:maven-shared-utils:jar:3.0.0:compile
[DEBUG]       org.codehaus.plexus:plexus-archiver:jar:2.9.1:compile
[DEBUG]          org.codehaus.plexus:plexus-io:jar:2.4.1:compile
[DEBUG] Created new class realm plugin>com.spotify:dockerfile-maven-plugin:1.3.0
[DEBUG] Importing foreign packages into class realm plugin>com.spotify:dockerfile-maven-plugin:1.3.0
[DEBUG]   Imported:  < maven.api
[DEBUG] Populating class realm plugin>com.spotify:dockerfile-maven-plugin:1.3.0
[DEBUG]   Included: com.spotify:dockerfile-maven-plugin:jar:1.3.0
[DEBUG]   Included: com.spotify:docker-client:jar:shaded:8.7.1
[DEBUG]   Included: com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:jar:2.8.8
[DEBUG]   Included: com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:jar:2.8.8
[DEBUG]   Included: com.fasterxml.jackson.core:jackson-core:jar:2.8.8
[DEBUG]   Included: com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.8.8
[DEBUG]   Included: com.fasterxml.jackson.datatype:jackson-datatype-guava:jar:2.8.8
[DEBUG]   Included: com.fasterxml.jackson.core:jackson-databind:jar:2.8.8
[DEBUG]   Included: com.fasterxml.jackson.core:jackson-annotations:jar:2.8.0
[DEBUG]   Included: org.glassfish.jersey.core:jersey-client:jar:2.22.2
[DEBUG]   Included: javax.ws.rs:javax.ws.rs-api:jar:2.0.1
[DEBUG]   Included: org.glassfish.jersey.core:jersey-common:jar:2.22.2
[DEBUG]   Included: javax.annotation:javax.annotation-api:jar:1.2
[DEBUG]   Included: org.glassfish.jersey.bundles.repackaged:jersey-guava:jar:2.22.2
[DEBUG]   Included: org.glassfish.hk2:osgi-resource-locator:jar:1.0.1
[DEBUG]   Included: org.glassfish.hk2:hk2-api:jar:2.4.0-b34
[DEBUG]   Included: org.glassfish.hk2:hk2-utils:jar:2.4.0-b34
[DEBUG]   Included: org.glassfish.hk2.external:aopalliance-repackaged:jar:2.4.0-b34
[DEBUG]   Included: org.glassfish.hk2.external:javax.inject:jar:2.4.0-b34
[DEBUG]   Included: org.glassfish.hk2:hk2-locator:jar:2.4.0-b34
[DEBUG]   Included: org.javassist:javassist:jar:3.18.1-GA
[DEBUG]   Included: org.glassfish.jersey.connectors:jersey-apache-connector:jar:2.22.2
[DEBUG]   Included: org.glassfish.jersey.media:jersey-media-json-jackson:jar:2.22.2
[DEBUG]   Included: org.glassfish.jersey.ext:jersey-entity-filtering:jar:2.22.2
[DEBUG]   Included: org.apache.commons:commons-compress:jar:1.9
[DEBUG]   Included: commons-io:commons-io:jar:2.5
[DEBUG]   Included: org.apache.httpcomponents:httpclient:jar:4.5
[DEBUG]   Included: commons-logging:commons-logging:jar:1.2
[DEBUG]   Included: commons-codec:commons-codec:jar:1.9
[DEBUG]   Included: org.apache.httpcomponents:httpcore:jar:4.4.5
[DEBUG]   Included: com.github.jnr:jnr-unixsocket:jar:0.8
[DEBUG]   Included: com.github.jnr:jnr-ffi:jar:2.0.3
[DEBUG]   Included: com.github.jnr:jffi:jar:1.2.9
[DEBUG]   Included: com.github.jnr:jffi:jar:native:1.2.9
[DEBUG]   Included: org.ow2.asm:asm:jar:5.0.3
[DEBUG]   Included: org.ow2.asm:asm-commons:jar:5.0.3
[DEBUG]   Included: org.ow2.asm:asm-analysis:jar:5.0.3
[DEBUG]   Included: org.ow2.asm:asm-tree:jar:5.0.3
[DEBUG]   Included: org.ow2.asm:asm-util:jar:5.0.3
[DEBUG]   Included: com.github.jnr:jnr-x86asm:jar:1.0.2
[DEBUG]   Included: com.github.jnr:jnr-constants:jar:0.8.7
[DEBUG]   Included: com.github.jnr:jnr-enxio:jar:0.9
[DEBUG]   Included: com.github.jnr:jnr-posix:jar:3.0.12
[DEBUG]   Included: commons-lang:commons-lang:jar:2.6
[DEBUG]   Included: org.bouncycastle:bcpkix-jdk15on:jar:1.52
[DEBUG]   Included: org.bouncycastle:bcprov-jdk15on:jar:1.52
[DEBUG]   Included: ch.qos.logback:logback-classic:jar:1.2.1
[DEBUG]   Included: ch.qos.logback:logback-core:jar:1.2.1
[DEBUG]   Included: com.google.auth:google-auth-library-oauth2-http:jar:0.6.0
[DEBUG]   Included: com.google.auth:google-auth-library-credentials:jar:0.6.0
[DEBUG]   Included: com.google.http-client:google-http-client:jar:1.19.0
[DEBUG]   Included: com.google.code.findbugs:jsr305:jar:2.0.1
[DEBUG]   Included: com.google.http-client:google-http-client-jackson2:jar:1.19.0
[DEBUG]   Included: com.google.guava:guava:jar:19.0
[DEBUG]   Included: com.spotify:dockerfile-maven-extension:jar:1.3.0
[DEBUG]   Included: javax.enterprise:cdi-api:jar:1.0
[DEBUG]   Included: javax.annotation:jsr250-api:jar:1.0
[DEBUG]   Included: org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.2
[DEBUG]   Included: org.apache.maven:maven-builder-support:jar:3.3.9
[DEBUG]   Included: org.eclipse.aether:aether-util:jar:1.0.2.v20150114
[DEBUG]   Included: com.google.inject:guice:jar:no_aop:4.0
[DEBUG]   Included: aopalliance:aopalliance:jar:1.0
[DEBUG]   Included: org.codehaus.plexus:plexus-interpolation:jar:1.21
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:3.0.22
[DEBUG]   Included: org.codehaus.plexus:plexus-component-annotations:jar:1.6
[DEBUG]   Included: org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3
[DEBUG]   Included: org.sonatype.plexus:plexus-cipher:jar:1.4
[DEBUG]   Included: org.apache.commons:commons-lang3:jar:3.4
[DEBUG]   Included: org.apache.maven:maven-archiver:jar:3.0.0
[DEBUG]   Included: org.apache.maven.shared:maven-shared-utils:jar:3.0.0
[DEBUG]   Included: org.codehaus.plexus:plexus-archiver:jar:2.9.1
[DEBUG]   Included: org.codehaus.plexus:plexus-io:jar:2.4.1
[DEBUG] Configuring mojo com.spotify:dockerfile-maven-plugin:1.3.0:build from plugin realm ClassRealm[plugin>com.spotify:dockerfile-maven-plugin:1.3.0, parent: sun.misc.Launcher$AppClassLoader@70dea4e]
[DEBUG] Configuring mojo 'com.spotify:dockerfile-maven-plugin:1.3.0:build' with basic configurator -->
[DEBUG]   (f) buildDirectory = /Users/myuser/Documents/Private/JavaProjects/myproject/target
[DEBUG]   (f) classifier = docker-info
[DEBUG]   (f) connectTimeoutMillis = 300000
[DEBUG]   (f) contextDirectory = /Users/myuser/Documents/Private/JavaProjects/myproject
[DEBUG]   (f) dockerInfoDirectory = /Users/myuser/Documents/Private/JavaProjects/myproject/target/docker
[DEBUG]   (f) finalName = myproject
[DEBUG]   (f) forceCreation = false
[DEBUG]   (f) noCache = false
[DEBUG]   (f) project = MavenProject: nl.bitrade:myproject:1.0.0.3-SNAPSHOT @ /Users/myuser/Documents/Private/JavaProjects/myproject/pom.xml
[DEBUG]   (f) pullNewerImage = true
[DEBUG]   (f) readTimeoutMillis = 300000
[DEBUG]   (f) repository = 805657053710.dkr.ecr.eu-west-1.amazonaws.com/myproject
[DEBUG]   (f) retryCount = 1
[DEBUG]   (f) session = org.apache.maven.execution.MavenSession@1c65121
[DEBUG]   (f) skip = false
[DEBUG]   (f) skipBuild = false
[DEBUG]   (f) tag = 1.0.0.3-SNAPSHOT
[DEBUG]   (f) testOutputDirectory = /Users/myuser/Documents/Private/JavaProjects/myproject/target/test-classes
[DEBUG]   (f) verbose = false
[DEBUG]   (f) writeTestMetadata = true
[DEBUG] -- end configuration --
[DEBUG] Using configfile: /Users/myuser/.docker/config.json
[DEBUG] Failed to load Google application default credentials
java.io.IOException: The Application Default Credentials are not available. They are available if running in Google Compute Engine. Otherwise, the environment variable GOOGLE_APPLICATION_CREDENTIALS must be defined pointing to a file defining the credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
	at com.google.auth.oauth2.DefaultCredentialsProvider.getDefaultCredentials(DefaultCredentialsProvider.java:111)
	at com.google.auth.oauth2.GoogleCredentials.getApplicationDefault(GoogleCredentials.java:103)
	at com.google.auth.oauth2.GoogleCredentials.getApplicationDefault(GoogleCredentials.java:76)
	at com.spotify.plugin.dockerfile.AbstractDockerMojo.googleContainerRegistryAuthSupplier(AbstractDockerMojo.java:448)
	at com.spotify.plugin.dockerfile.AbstractDockerMojo.createRegistryAuthSupplier(AbstractDockerMojo.java:405)
	at com.spotify.plugin.dockerfile.AbstractDockerMojo.openDockerClient(AbstractDockerMojo.java:386)
	at com.spotify.plugin.dockerfile.AbstractDockerMojo.tryExecute(AbstractDockerMojo.java:219)
	at com.spotify.plugin.dockerfile.AbstractDockerMojo.execute(AbstractDockerMojo.java:208)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:47)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
[DEBUG] /Users/myuser/.docker/ca.pem, /Users/myuser/.docker/key.pem or /Users/myuser/.docker/cert.pem does not exist, not using SSL
[INFO] Building Docker context /Users/myuser/Documents/Private/JavaProjects/myproject
[INFO] 
[INFO] Image will be built as xxxxxxxxxxxxxx.dkr.ecr.eu-west-1.amazonaws.com/myproject:1.0.0.3-SNAPSHOT
[INFO] 
[INFO] Step 1/10 : FROM java:8
[INFO] Pulling from library/java
[INFO] Digest: sha256:c1ff613e8ba25833d2e1940da0940c3824f03f802c449f3d1815a66b7f8c0e9d
[INFO] Status: Image is up to date for java:8
[INFO]  ---> d23bdf5b1b1b
[INFO] Step 2/10 : LABEL vendor "MyProject"
[INFO]  ---> Using cache
[INFO]  ---> 7bfdfcaddae9
[INFO] Step 3/10 : EXPOSE 8080
[INFO]  ---> Using cache
[INFO]  ---> a75973c4f510
[INFO] Step 4/10 : HEALTHCHECK CMD curl --fail http://localhost:8080/health || exit 1
[INFO]  ---> Using cache
[INFO]  ---> 1af7c702980e
[INFO] Step 5/10 : ENV JAVA_TOOL_OPTIONS -Dfile.encoding=UTF-8
[INFO]  ---> Using cache
[INFO]  ---> ed92aa96b020
[INFO] Step 6/10 : VOLUME /tmp
[INFO]  ---> Using cache
[INFO]  ---> db50832836e7
[INFO] Step 7/10 : ADD target/myproject.jar /myproject.jar
[INFO]  ---> 92f84b9dbbba
[INFO] Removing intermediate container 862187e75fb8
[INFO] Step 8/10 : RUN bash -c 'touch /myproject.jar'
[INFO]  ---> Running in 9ed3f720bf56
[INFO]  ---> a0fe406755eb
[INFO] Removing intermediate container 9ed3f720bf56
[INFO] Step 9/10 : RUN apt-get update && apt-get install -y         curl
[INFO]  ---> Running in be0ad24735ed
[INFO] Get:1 http://security.debian.org jessie/updates InRelease [63.1 kB]
[INFO] Get:2 http://security.debian.org jessie/updates/main amd64 Packages [524 kB]
[INFO] Ign http://deb.debian.org jessie InRelease
[INFO] Get:3 http://deb.debian.org jessie-updates InRelease [145 kB]
[INFO] Get:4 http://deb.debian.org jessie-backports InRelease [166 kB]
[INFO] Get:5 http://deb.debian.org jessie Release.gpg [2373 B]
[INFO] Get:6 http://deb.debian.org jessie Release [148 kB]
[INFO] Get:7 http://deb.debian.org jessie-updates/main amd64 Packages [17.8 kB]
[INFO] Get:8 http://deb.debian.org jessie-backports/main amd64 Packages [1175 kB]
[INFO] Get:9 http://deb.debian.org jessie/main amd64 Packages [9065 kB]
[INFO] Fetched 11.3 MB in 10s (1071 kB/s)
[INFO] Reading package lists...
[INFO] 
[INFO] Reading package lists...
[INFO] 
[INFO] Building dependency tree...
[INFO] 
[INFO] Reading state information...
[INFO] 
[INFO] curl is already the newest version.
[INFO] 0 upgraded, 0 newly installed, 0 to remove and 41 not upgraded.
[INFO]  ---> 3c391e69b61e
[INFO] Removing intermediate container be0ad24735ed
[INFO] Step 10/10 : ENTRYPOINT java -Djava.security.egd=file:/dev/./urandom -jar /myproject.jar
[INFO]  ---> Running in f3f49ebf6af6
[INFO]  ---> 47bb7473eead
[INFO] Removing intermediate container f3f49ebf6af6
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
...
[INFO] MyProject .......................................... FAILURE [ 27.876 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 39.078 s
[INFO] Finished at: 2017-06-29T18:26:57+02:00
[INFO] Final Memory: 62M/511M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.spotify:dockerfile-maven-plugin:1.3.0:build (default) on project myproject: Execution default of goal com.spotify:dockerfile-maven-plugin:1.3.0:build failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.spotify:dockerfile-maven-plugin:1.3.0:build (default) on project myproject: Execution default of goal com.spotify:dockerfile-maven-plugin:1.3.0:build failed.
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:47)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal com.spotify:dockerfile-maven-plugin:1.3.0:build failed.
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 26 more
Caused by: java.lang.NullPointerException
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:212)
	at com.google.common.base.Splitter.split(Splitter.java:407)
	at com.spotify.plugin.dockerfile.LoggingProgressHandler.handleGeneric(LoggingProgressHandler.java:83)
	at com.spotify.plugin.dockerfile.LoggingProgressHandler.progress(LoggingProgressHandler.java:67)
	at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1389)
	at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1348)
	at com.spotify.plugin.dockerfile.BuildMojo.buildImage(BuildMojo.java:157)
	at com.spotify.plugin.dockerfile.BuildMojo.execute(BuildMojo.java:95)
	at com.spotify.plugin.dockerfile.AbstractDockerMojo.tryExecute(AbstractDockerMojo.java:219)
	at com.spotify.plugin.dockerfile.AbstractDockerMojo.execute(AbstractDockerMojo.java:208)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	... 27 more
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :myproject
```

**Comment by bkmeneguello** [comment]
I'm with the same error here

> docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 29
Server Version: 17.06.0-ce

> Execution docker-build of goal com.spotify:dockerfile-maven-plugin:1.3.0:build failed.
 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:225)
 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
 	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:320)
 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.lang.reflect.Method.invoke(Method.java:498)
 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:409)
 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:352)
 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.lang.reflect.Method.invoke(Method.java:498)
 	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:39)
 	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:122)
 	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:50)
 Caused by: org.apache.maven.plugin.PluginExecutionException: Execution docker-build of goal com.spotify:dockerfile-maven-plugin:1.3.0:build failed.
 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:110)
 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
 	... 26 more
 Caused by: java.lang.NullPointerException
 	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:212)
 	at com.google.common.base.Splitter.split(Splitter.java:407)
 	at com.spotify.plugin.dockerfile.LoggingProgressHandler.handleGeneric(LoggingProgressHandler.java:83)
 	at com.spotify.plugin.dockerfile.LoggingProgressHandler.progress(LoggingProgressHandler.java:67)
 	at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1389)
 	at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1348)
 	at com.spotify.plugin.dockerfile.BuildMojo.buildImage(BuildMojo.java:157)
 	at com.spotify.plugin.dockerfile.BuildMojo.execute(BuildMojo.java:95)
 	at com.spotify.plugin.dockerfile.AbstractDockerMojo.tryExecute(AbstractDockerMojo.java:219)
 	at com.spotify.plugin.dockerfile.AbstractDockerMojo.execute(AbstractDockerMojo.java:208)
 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
 	... 27 more

**Comment by mar-io** [comment]
I think we are all waiting on this pull request to be merged and release: https://github.com/spotify/dockerfile-maven/pull/30

I just reverted back to "17.03.1-ce-mac12". Had the dmg on my computer thankfully.

**Comment by davidxia** [comment]
#30 has been merged. I'm trying to figure out how to release this though.

In the meantime for mac, if you use docker-machine you can create VMs with dockers of any version.

```
docker-machine -D create -d virtualbox --virtualbox-boot2docker-url https://github.com/boot2docker/boot2docker/releases/download/v17.03.1-ce/boot2docker.iso --virtualbox-cpu-count 2 --virtualbox-memory 2048 17.03.1
```

[More releases here.](https://github.com/boot2docker/boot2docker/releases)

**Comment by beihaifeiwu** [comment]
I got the flow exception on windows when docker upgrade:
```bash
[INFO] --- dockerfile-maven-plugin:1.3.0:build (default) @ nagrand-ws ---
[INFO] Building Docker context F:\Source\nagrand\nagrand-ws
[INFO]
[INFO] Image will be built as 10.0.25.21:5000/nagrand-ws:2.1-SNAPSHOT
[INFO]
[WARNING] An attempt failed, will retry 1 more times
org.apache.maven.plugin.MojoExecutionException: Could not build image
        at com.spotify.plugin.dockerfile.BuildMojo.buildImage(BuildMojo.java:164)
        at com.spotify.plugin.dockerfile.BuildMojo.execute(BuildMojo.java:95)
        at com.spotify.plugin.dockerfile.AbstractDockerMojo.tryExecute(AbstractDockerMojo.java:219)
        at com.spotify.plugin.dockerfile.AbstractDockerMojo.execute(AbstractDockerMojo.java:208)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: com.spotify.docker.client.exceptions.DockerException: com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.JsonMappingException: Can not construct instance of com.spotify.docker.client.messages.RegistryAuth: no String-argument constructor/factory method to deserialize from String value ('wincred')
 at [Source: N/A; line: -1, column: -1] (through reference chain: java.util.LinkedHashMap["credsStore"])
        at com.spotify.docker.client.auth.ConfigFileRegistryAuthSupplier.authForBuild(ConfigFileRegistryAuthSupplier.java:96)
        at com.spotify.docker.client.auth.MultiRegistryAuthSupplier.authForBuild(MultiRegistryAuthSupplier.java:77)
        at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1371)
        at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1348)
        at com.spotify.plugin.dockerfile.BuildMojo.buildImage(BuildMojo.java:157)
        ... 25 more
Caused by: com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.JsonMappingException: Can not construct instance of com.spotify.docker.client.messages.RegistryAuth: no String-argument constructor/factory method to deserialize from String value ('wincred')
 at [Source: N/A; line: -1, column: -1] (through reference chain: java.util.LinkedHashMap["credsStore"])
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:270)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.DeserializationContext.instantiationException(DeserializationContext.java:1456)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1012)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.deser.ValueInstantiator._createFromStringFallbacks(ValueInstantiator.java:370)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromString(StdValueInstantiator.java:315)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromString(BeanDeserializerBase.java:1283)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:159)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:150)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap(MapDeserializer.java:517)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:362)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:27)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1187)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:3770)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2099)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.ObjectMapper.treeToValue(ObjectMapper.java:2596)
        at com.spotify.docker.client.DockerConfigReader.parseDockerConfig(DockerConfigReader.java:94)
        at com.spotify.docker.client.DockerConfigReader.fromConfig(DockerConfigReader.java:48)
        at com.spotify.docker.client.auth.ConfigFileRegistryAuthSupplier.authForBuild(ConfigFileRegistryAuthSupplier.java:94)
        ... 29 more
[INFO] Building Docker context F:\Source\nagrand\nagrand-ws
```

**Comment by davidxia** [comment]
Fixed by https://github.com/spotify/dockerfile-maven/pull/30 and relased as 1.3.1.

@beihaifeiwu This is a [docker-client](https://github.com/spotify/docker-client) issue. Can you create a separate issue there?

**Comment by davidxia** [comment]
@beihaifeiwu There's an [existing issue here](https://github.com/spotify/docker-client/issues/657#issuecomment-305577737).

**Comment by tstroband** [comment]
Works great!
Thanks a lot!!!
